integrationTests/cypress/support/commands/misc_commands.js

```diff
@@ -0,0 +1,109 @@
+// ***********************************************
+// commands for the login/logout flow
+// ***********************************************
+let doguName = "redmine"
+
+/**
+ * Log the test user in the cas.
+ * @param {String} username - The username of the user.
+ * @param {String} password - The password for the user.
+ */
+const login = (username, password) => {
+    cy.visit("/" + doguName)
+
+    cy.clickWarpMenuCheckboxIfPossible()
+
+    cy.get('input[name="username"]').type(username)
+    cy.get('input[name="password"]').type(password)
+    cy.get('button[name="submit"]').click()
+}
+
+
+/**
+ * Log the admin user defined in the ces_admin_data.json in the cas.
+ */
+const loginAdmin = () => {
+    cy.fixture("ces_admin_data").then(function (admin) {
+        cy.visit("/" + doguName)
+
+        cy.clickWarpMenuCheckboxIfPossible()
+
+        cy.get('input[name="username"]').type(admin.username)
+        cy.get('input[name="password"]').type(admin.password)
+        cy.get('button[name="submit"]').click()
+    })
+}
+
+/**
+ * Log the current user out of the cas via back-channel logout.
+ */
+const logout = () => {
+    cy.visit("/cas/logout")
+}
+
+/**
+ * Handles the warp menu tooltip by clicking the 'do not show again' checkbox on the first time.
+ */
+const clickWarpMenuCheckboxIfPossible = () => {
+    cy.get('div[id="warp-menu-container"]').then(function (container) {
+        let warpContainer = container.children( ".warp-menu-column-tooltip")
+        if (warpContainer.length == 1) {
```

Comment:
MagicNumber Code smell, why is this lenght == 1. Can it be something else or is there another attribute of the warp menu which can be tested in a more safer way?

Reply:
Cypress does not support fail states. In this case, it is only possible to check whether the child of our container has the length of one. This indicates for us that the warp menu dialog is open and can be toggled via the toggle. It is kinda like a null pointer check for Cypress